### DISCUSSION TITLE
webworker中使用offscreenCanvas初始化粒子动效引擎报错

### DISCUSSION THREAD

**Discussion by viviuolo** [galacean/discussion#105] [Engine]
主线程：
const offscreenCanvas = document.createElement('canvas');
const offCanvas = offscreenCanvas.transferControlToOffscreen();
const worker = new Worker('worker.js');
worker.postMessage({
      type: 'init',
      data: {
          offCanvas
      }
  }, [offCanvas]) 

worker：
const engine = await WebGLEngine.create({ canvas: offCanvas });
......
// texture为base64图片
engine.resourceManager
    .load({
        url: texture,
        type: AssetType.Texture2D
    }) .then((resource) => {});
错误信息：
![image](https://github.com/galacean/discussion/assets/22727579/d8f44419-846f-4159-8465-45187697e2f8)
![image](https://github.com/galacean/discussion/assets/22727579/97b0f2b0-4e27-4aa2-b696-41578f6f52a5)

**Comment by eyworldwide** [comment]
@viviuolo  引擎对 [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) 还没做过适配，有兴趣的话欢迎给我们 Pull Request 支持这个 API ^-^
